packages/contracts/tasks/l2-gasprice.ts

```diff
@@ -0,0 +1,64 @@
+/* Imports: External */
+import { ethers } from 'ethers'
+import { task } from 'hardhat/config'
+import * as types from 'hardhat/internal/core/params/argumentTypes'
+
+import { predeploys } from '../src/predeploys'
+import { getContractDefinition } from '../src/contract-defs'
+
+task('set-l2-gasprice')
+  .addOptionalParam('l2GasPrice', 'Gas Price to set on L2', 0, types.int)
+  .addOptionalParam(
+    'transactionGasPrice',
+    'Gas price to set on L2 transaction',
+    undefined,
+    types.int
+  )
+  .addOptionalParam(
+    'contractsRpcUrl',
+    'Sequencer HTTP Endpoint',
+    process.env.CONTRACTS_RPC_URL,
+    types.string
+  )
+  .addOptionalParam(
+    'contractsDeployerKey',
+    'Private Key',
+    process.env.CONTRACTS_DEPLOYER_KEY,
+    types.string
+  )
+  .setAction(async (args, hre: any, runSuper) => {
+    const provider = new ethers.providers.JsonRpcProvider(args.contractsRpcUrl)
+    const signer = new ethers.Wallet(args.contractsDeployerKey).connect(
+      provider
+    )
+
+    const GasPriceOracleArtifact = getContractDefinition(
+      'OVM_GasPriceOracle',
+      true
+    )
+
+    const GasPriceOracle = new ethers.Contract(
+      predeploys.OVM_GasPriceOracle,
+      GasPriceOracleArtifact.abi,
+      provider
+    )
+
+    const addr = await signer.getAddress()
+    console.log(`Using signer ${addr}`)
+    const owner = await GasPriceOracle.callStatic.owner()
```

Comment:
Hmmm do we use this in prod? A little sus if the owner of the GPO is just a private key sitting on a machine somewhere...

Reply:
See the comment in the PR message above: `This shouldn't be used in production but is useful for testnet/local deployments.`

Reply:
#1103 is what will be used in prod